### PR TITLE
Enable CRC greentea test

### PR DESCRIPTION
**Pull request by gpsimenos**
This PR enables the CRC greentea test. The test can be built and run with the existing steps in README.

Accompanies https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/pull/12, to be merged after #12.

### CODE REVIEW

**Comment by MarceloSalazar** [comment]
Looks good.
As https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/pull/12 is merged, we should update the submodule references to the MCU-Driver-HAL repo.
Additionally, please share here the Greentea test logs.

**Comment by gpsimenos** [comment]
@ithinuel would you be able to paste your greentea logs from the CRC test here?

**Comment by pfalcon-linaro** [comment]
@MarceloSalazar, @gpsimenos:

> As MCU-Driver-HAL/MCU-Driver-HAL#12 is merged, we should update the submodule references to the MCU-Driver-HAL repo.

I'm trying to figure out how to solve similar issue for my PRs https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/pull/19, https://github.com/MCU-Driver-HAL/MCU-Driver-ST/pull/13 .

Any suggestions on the "best practices" for that? As far as I can see, initially this PR didn't contain any submodule update at all, then MCU-Driver-HAL's counterpart was merged first, then commit to reference updated MCU-Driver-HAL revision was added here.

The issue is that it doesn't scale to automated CI - there's no (immediate) way how to test PR here until MCU-Driver-HAL's part is merged, but we can't be sure it's correct until it's (at least) built, which requires complete PR in this repo.

Indeed, that looks like the reason why some projects might quit git submodules and devise some adhoc solution which is supposedly easier to use for such cross-repo cases ;-). For example, in zephyr's west manifest it's possible to initially reference "the other PR" as `pull/NNN/head`, have that be built by CI, then when "the other PR" is merged, replace it with the actual commit id/tag. 

But it should be (kinda) possible to do the same with git submodules too, we just need to try, and maybe add some helper scripts/instructions to make that easy.

**Comment by pfalcon-linaro** [comment]
> I'm trying to figure out ...

Reposted as a separate ticket: https://github.com/MCU-Driver-HAL/MCU-Driver-ST/issues/14

File: sdfx_st/tests/mbed_hal/crc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
+
+set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../MCU-Driver-HAL CACHE INTERNAL "")
+set(MBED_CONFIG_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../.mbedbuild CACHE INTERNAL "")
+
+set(TEST_TARGET sdfx-st-test-crc)
```
Comment (ithinuel) [inline sdfx_st/tests/mbed_hal/crc/CMakeLists.txt:21]: I find this convoluted.

Why not using those value directly?

File: sdfx_st/tests/mbed_hal/crc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+include(${MBED_PATH}/tools/cmake/app.cmake)
+
+add_subdirectory(../../../.. ./MCU-Driver-ST)
+
+add_executable(${TEST_TARGET})
+
+mbed_configure_app_target(${TEST_TARGET})
+
+project(${TEST_TARGET})
```
Comment (ithinuel) [inline sdfx_st/tests/mbed_hal/crc/CMakeLists.txt:32]: This like should be 
> Call the project() command near the top of the top-level CMakeLists.txt, but after calling cmake_minimum_required()

according to cmake's [doc](https://cmake.org/cmake/help/latest/command/project.html)

it should be moved around line 17 from this file.

**Review by ithinuel — CHANGES_REQUESTED** [review]

**Comment by MarceloSalazar** [comment]
@gpsimenos let us know once the PR is ready for review

**Comment by gpsimenos** [comment]
> @gpsimenos let us know once the PR is ready for review

It's ready for review. Will update MCU-Driver-HAL reference once approved.

File: sdfx_st/tests/mbed_hal/crc/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+mbed_configure_app_target(sdfx-st-test-crc)
+
+target_link_libraries(sdfx-st-test-crc
+    PRIVATE
+        sdfx-test-crc
+)
+
+mbed_set_post_build(sdfx-st-test-crc)
+
+option(VERBOSE_BUILD "Have a verbose build process")
```
Comment (ithinuel) [inline sdfx_st/tests/mbed_hal/crc/CMakeLists.txt:38]: Can we remove that and what follows please?

**Review by ithinuel — CHANGES_REQUESTED** [review]

File: sdfx_st/tests/mbed_hal/crc/CMakeLists.txt (hunk 3 above)
Comment (MarceloSalazar) [reply sdfx_st/tests/mbed_hal/crc/CMakeLists.txt:38]: This is needed to be aligned with https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/pull/43

**Review by ithinuel — DISMISSED** [review]

**Comment by gpsimenos** [comment]
MCU-Driver-HAL reference updated, ready for merge

**Review by MarceloSalazar — APPROVED** [review]

**Pull request merged**